source/wp-content/themes/wporg-showcase-2022/functions.php

```diff
@@ -316,3 +317,46 @@ function document_title( $parts ) {
 function document_title_separator( $title ) {
 	return '&#124;';
 }
+
+/**
+ * Add meta tags for richer social media integrations.
+ */
+function add_social_meta_tags() {
+	$default_image = get_stylesheet_directory_uri() . '/images/social-image.png';
+	$site_title    = function_exists( '\WordPressdotorg\site_brand' ) ? \WordPressdotorg\site_brand() : 'WordPress.org';
+	$og_fields = [
+		'og:title'       => wp_get_document_title(),
+		'og:description' => __( 'Discover inspiration in some of the most beautiful, best designed WordPress websites.', 'wporg' ),
+		'og:site_name'   => $site_title,
+		'og:type'        => 'website',
+		'og:url'         => home_url(),
+		'og:image'       => esc_url( $default_image ),
+	];
+
+	if ( is_tag() || is_category() ) {
+		$og_fields['og:url'] = esc_url( get_term_link( get_queried_object_id() ) );
+	} elseif ( is_single() ) {
+		$og_fields['og:description'] = strip_tags( get_the_excerpt() );
+		$og_fields['og:url']         = esc_url( get_permalink() );
+		$og_fields['og:image']       = esc_url( site_screenshot_src( get_post() ) );
+	}
+
+	printf( '<meta name="twitter:card" content="summary_large_image">' . "\n" );
+	printf( '<meta name="twitter:site" content="@WordPress">' . "\n" );
+	printf( '<meta name="twitter:image" content="%s" />' . "\n", esc_url( $og_fields['og:image'] ) );
+
+	foreach ( $og_fields as $property => $content ) {
+		printf(
+			'<meta property="%1$s" content="%2$s" />' . "\n",
+			esc_attr( $property ),
+			esc_attr( $content )
+		);
+	}
+
+	if ( isset( $og_fields['og:description'] ) ) {
```

Comment:
I'm not sure if this should be outputting both `<meta property="og:description"` and `<meta name="description"` I thought it should only be the latter.. But both works for me if we do that elsewhere.

Reply:
We do it in the loop above. 👍.

Reply:
Yeah, I was saying I'm not sure it should output both.

We output both on plugins, so this is fine.
```
$ curl https://wordpress.org/plugins/hello-dolly/ -s | grep '<meta.\+description'
<meta property="og:description" content="This is not just a plugin, it symbolizes the hope and enthusiasm of an entire generation summed up in two words sung most famously by Louis Armstrong." />
<meta name="description" content="This is not just a plugin, it symbolizes the hope and enthusiasm of an entire generation summed up in two words sung most famously by Louis Armstrong." />
```